garden-backend-service/src/api/routes/garden_search_record.py

```diff
@@ -1,17 +1,20 @@
-import asyncio
-
 from fastapi import APIRouter, Depends, exceptions, status
 from globus_sdk import SearchClient
 from src.api.dependencies.auth import AuthenticationState, authenticated
 from src.api.dependencies.search import get_globus_search_client
-from src.api.routes._utils import is_doi_registered
+from src.api.routes._utils import deprecated, is_doi_registered, poll_globus_search_task
 from src.api.schemas.search import DeleteSearchRecordRequest, PublishSearchRecordRequest
 from src.config import Settings, get_settings
 
 router = APIRouter(prefix="/garden-search-record")
 
 
-@router.post("", status_code=status.HTTP_200_OK)
+@router.post("", status_code=status.HTTP_200_OK, deprecated=True)
```

Comment:
I don't think `deprecated=True` is what we want here, since we don't actually want these routes to be deprecated until we flip the switch on SYNC_SEARCH_INDEX. That way, if we discover a bug and need to un-deprecate the routes we can just change the environment variable instead of needing to change the code and redeploy.

Reply:
thinking through this a little I think the easiest way to do this is just check the flag in the body of the route to raise the deprecation error. but if there's a way to capture that with your decorator that could be cool too. I'd bet that naively changing this to `deprecated=get_settings().SYNC_SEARCH_INDEX` would cause problems with tests/ mocked settings but I haven't checked.